packages/treasury/src/collectRewardFees.js

```diff
@@ -17,16 +16,17 @@ export const makeMakeCollectFeesInvitation = (
     const { zcfSeat: transferSeat } = zcf.makeEmptySeatKit();
     await E.get(offerTo(zcf, invitation, {}, {}, transferSeat)).deposited;
 
-    const totalTransferred = AmountMath.add(
-      feeSeat.getAmountAllocated('RUN', runBrand),
-      transferSeat.getAmountAllocated('RUN', runBrand),
+    feeSeat.decrementBy(
+      seat.incrementBy({ RUN: feeSeat.getAmountAllocated('RUN', runBrand) }),
```

Comment:
See https://github.com/Agoric/agoric-sdk/pull/3224

Reply:
> For each direction, what happens if seat === feeSeat ? Always important to think about these aliasing cases, either to ensure they cannot happen or to check that they would do the right thing.

Good question. If `seat === feeSeat`, then (assuming that the seat has enough allocated for the decrement to succeed), a staged allocation is created, but it is equal to the current allocation. If the seat does not have enough allocated for a decrement to succeed, this results in an error being thrown, as it would generally for trying to subtract more than exists.

A mistaken `seat === feeSeat` will be caught in the `reallocate` call when we check for seat aliasing. I think this is the correct place, because conceptually, `zcfSeat.incrementBy` and `zcfSeat.decrementBy` have no concept of other seats, and seats are only related to each other at the point of the `reallocate` call.

Reply:
I like this answer. Resolving.